pallets/block-reward/src/lib.rs

```diff
@@ -257,17 +311,39 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Sets the default averaging-function by AverageSelector.
+		#[pallet::weight(T::WeightInfo::set_averaging_function_selector())]
+		pub fn set_averaging_function_selector(
+			origin: OriginFor<T>,
+			avg_sel: AverageSelector,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			AverageSelectorConfig::<T>::put(avg_sel);
+
+			Self::deposit_event(Event::<T>::AverageSelectorChanged(avg_sel));
+
+			Ok(().into())
+		}
 	}
 
 	impl<Moment, T: Config> OnTimestampSet<Moment> for Pallet<T> {
 		fn on_timestamp_set(_moment: Moment) {
-			if T::Currency::total_issuance() >= Self::max_currency_supply() {
-				return
-			}
-
-			let inflation = T::Currency::issue(Self::block_issue_reward());
-			let value = inflation.peek();
-			Self::distribute_imbalances(inflation, Event::<T>::BlockRewardsDistributed(value));
+			let txfees = TokenLocker::<T>::mutate(|lock| {
+				let locked = *lock;
+				*lock = BalanceOf::<T>::zero();
+				T::Currency::issue(locked)
```

Comment:
So, we only issue new tokens if `total_supply` is lower than `max_currency_supply`. This `issue()`-call was only because of technical reasons, but I don't rember right now. I will check it again, if a transfer is possible :+1: because this is like a transfer of tokens out of the TokenLocker again...